apis/refs/v1beta1/secret/basicauth.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secret
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ SecretRef = &BasicAuthSecret{}
+
+type BasicAuthSecret struct {
+	// +required
+	// The `metadata.name` field of a Kubernetes `Secret`
+	Name string `json:"name,omitempty"`
+	// The `metadata.namespace` field of a Kubernetes `Secret`.
+	Namespace string `json:"namespace,omitempty"`
+
+	// The public field with json:"-" tag is to skip the field
+	// in the CRD, and bypass "the unexported field error"
```

Comment:
Agree to not export (that's also what I did at the beginning). But when this object is referenced and parsed in `runtime.DefaultUnstructuredConverter.FromUnstructured`, it panics. Here's the related k/k issue.  
Here's the related issue https://github.com/kubernetes/kubernetes/issues/124154

Reply:
Ah - good to know.  Disappointing that we haven't fixed that!

Reply:
I guess we can image that we _might_ let users set the raw values directly, and then having these fields makes sense.

We don't want to let users set the raw values, because we want to nudge them to using a secret.  But if we get push-back, we could change our minds here and allow the values to be set via the json tag.  And then if we did that, this structure and normalization _would_ make sense.